fix(gemini-streaming):  AssertionError (Issue#2626)

binaryCrossEntropy opened the pull request.

DouweM commented:
@binaryCrossEntropy Thank you! Can you see if you can add a test that triggers this code path? Likely just using the code execution built-in tool will do it.

binaryCrossEntropy commented:
Hiya @DouweM - I’ve added a test that triggers the path where Gemini streams executable_code / code_execution_result.
I put my test case in `tests/models/test_google.py` and also added a cassette file.

Files modified/added:

- `Test: tests/models/test_google.py::test_google_model_builtin_code_execution_stream`

- `Cassette: tests/fixtures/cassettes/test_google_model_builtin_code_execution_stream.yaml`

The test passes with this fix, and it fails with an AssertionError if I remove the code.

<img width="1206" height="298" alt="Image" src="https://github.com/user-attachments/assets/226e440f-da94-426a-9013-1e8b8355b706" />

Test pass:
<img width="797" height="177" alt="test_pass" src="https://github.com/user-attachments/assets/a1772167-9b22-4ed3-8b1c-d65d7e08680f" />

Happy to work on other feedback :)
